ironfish/src/consensus/consensus.ts

```diff
@@ -33,6 +33,12 @@ export type ConsensusParameters = {
    * The minimum fee that a transaction must have to be accepted
    */
   minFee: number
+
+  /**
+   * Before upgrade we had negative block mine time in block headers. After this
+   * block we do a check to disallow negative block mine time.
+   */
+  disallowNegativeBlockMineTime: number
```

Comment:
I don't think we have a concept of block mined time. Just block timestamps. I know it seems pedantic, but I would recommend referring to concepts that can be found in our models like `block.header.timestamp`.

Reply:
I am not inclined to name is disallowNegativeBlockTimestamp since the timestamp itself is never negative, its the comparison with the previous block timestamp.
How about `disallowBlockTimestampSmallerThanPreviousBlock`? I know its pretty long but I think thats accurate if we dont want to introduce block mine time to codbase and just use block timestamp

Reply:
Some proposals: `disallowPastBlockTimestamp` or `enforceMonotonicBlockTimestamp`

Reply:
"Monotonic block timestamp(s)" is probably the perfect term for this project too

Reply:
I like the proposal, I will rename to `enforceSequentialBlockTime`